Added options in New Themes subdialog for frequency and summary tables.

anastasia-mbithe opened the pull request:
Fixes #7993

I have added controls, parameters, and functions (set R code) needed for the new Themes sub dialog. However, the subject requires more discussion with @lilyclements and @N-thony.

anastasia-mbithe commented:
@rdstern, Yes I had a discussion with @lilyclements early this week and I have already resumed working on this.

anastasia-mbithe commented:
@rdstern This is almost ready and its already working, so testing can be done. (I have been testing through the summary tables dialog). 
@lilyclements you can check if there are other more features to add, apart from footnotes and source notes parameters which I should be done by Friday hopefully.

rdstern reviewed (commented):
@anastasia-mbithe this is starting to look great.
a) I tried the built-in themes.  They always set back to the first, namely Dark, each time you re-open the sub-dialogue.  And I think the NYTimes theme always gives the Dark theme.  The others seem to work.
b) The defining your own theme is starting to look very comprehensive.  I am assuming this is still work in progress, at least on some of the names for the rows, etc.
c) So tell me when you want me to look more carefully.
d) And separately to that, the reordering of the summaries still doesn't work.  And there will need to be a tab in the Format table to be able to change the names of the summaries.

lilyclements commented:
@anastasia-mbithe are you happy to work on the changes that @rdstern has outlined above, or did you want me to review as well before you work on the changes (just let me know whichever suits you best)

anastasia-mbithe commented:
@rdstern, I checked the ucrReorder issue with @N-thony  and it seems it's not an issue from the control since it passes the summaries in the order defined by the user. The issue is from the Rcode/Function which gives the output in alphabetical order. @lilyclements, can also check to confirm that, and see if it's within our power or the package developer's to fix it.

rdstern commented:
@anastasia-mbithe I hope the order of the summaries can soon be solved.

I also wonder whether the themes progress should now be merged, and then you are free to start on some of the remaining tabs.

I have also checked on the possibility of changing the names of the summaries.  David says that should be easy to do as the summaries are produced, i.e. before using mmtable2.  That would then be a new tab in the Prepare > Data reshape > Summaries dialogue and could then be added to the tables dialogue.  Perhaps make a new issue of this?  It is not as urgent as adding the other features from gt and gtExtras to the Format Tables dialogue.

anastasia-mbithe commented:
> @anastasia-mbithe I hope the order of the summaries can soon be solved.
> 
> I also wonder whether the themes progress should now be merged, and then you are free to start on some of the remaining tabs.
> 
> I have also checked on the possibility of changing the names of the summaries. David says that should be easy to do as the summaries are produced, i.e. before using mmtable2. That would then be a new tab in the Prepare > Data reshape > Summaries dialogue and could then be added to the tables dialogue. Perhaps make a new issue of this? It is not as urgent as adding the other features from gt and gtExtras to the Format Tables dialogue.

Yes, this is ready for testing, the themes bit, we've disabled some tabs like the row/columns/stub, etc since they are not in use at the moment. Until we implement the remaining tabs in the format sub dialog.

anastasia-mbithe commented:
@rdstern, since the Format sub dialog has only the Titles and Themes tabs implemented, and the Titles' has footnotes and source notes too, you can test the themes related to titles, footnotes, and source notes.
On the Themes sub dialog, we will enable columns, rows, and stub tabs after the Facets, Rows, and columns tabs on the Format summary are implemented.
Some of the gtExtras Themes are included here
![image](https://user-images.githubusercontent.com/89011047/221114051-f49e3517-0718-483b-93fd-6216cafabece.png)


I had a discussion with Lily yesterday evening and she will sort out the summary reordering next week, probably on Thursday.
There's more still to be done on these sub-dialogs.

rdstern commented:
@anastasia-mbithe thank you for the update.

As you have been working on themes, I wonder about the theme I mentioned above, and also about including the gtsummary package, so we can have that theme?

I alsao wonder what else you are working on now, while waiting for the inputs on the tables from @lilyclements ?  Should you be asking @N-thony for other tasks?  There is always plenty still to do!

anastasia-mbithe commented:
@lilyclements, did you mean the factors multiple receiver?
![image](https://user-images.githubusercontent.com/89011047/224247217-ba1b281d-7d58-4219-9422-51dbf11d1e20.png)

Yes it should be filled. It had a minor bug of not retaining the variables once you reopen the dialog, but that is fixed in this PR now.

lloyddewit commented:
@anastasia-mbithe I'm not completely clear about the status of this PR. In your view, is it working correctly and ready for testing? Or do you still wish to make changes?
thanks for clarifying

anastasia-mbithe commented:
@lloyddewit for now, this is ready for testing.

In summary, the following have been implemented in this pr;
1) Fixed the bug on the Factors receiver
2) On sdgFormat summary:- 
     Changed the tabs to Titles, Facets,Rows,Columns and Themes. For now, Facets, rows and columns have been disabled since they are yet to be implemented (an issue  to do with the functions, Lily is following up with mmtable package developer)
3) On the Themes; Select Themes uses the gtExtra package themes.
        For the Manual Theme; all the tabs have been implemented, but the Columns, rows, and stub have been disabled for now since their corresponding functions in sdgFormat summary are yet to be implemented.
4) Lily has fixed the issue of reordering summaries in pr #8211.

I hope this will make it easier when reviewing.

lloyddewit commented:
@anastasia-mbithe do you still expect changes to this PR from yourself or others? Or is this PR ready for testing? 
Thanks for clarifying

anastasia-mbithe commented:
@lloyddewit This is ready for testing.

lloyddewit commented:
@rdstern please could you test? thanks

rdstern reviewed (approved):
@anastasia-mbithe looks good.  And I really like the parallel between themes for ggplot and those for gt.

lloyddewit commented on instat/sdgSummaryThemes.vb:
```diff
@@ -17,7 +17,1080 @@
+        ucrChkFootnotesMarks.AddParameterPresentCondition(False, "footnotes.marks", False)
+
+        ucrChkFootnotesMultiline.SetText("Footnotes Multiline")
+        ucrInputFootnotesMultiline.SetParameter(New RParameter("footnotes.marks"))
+        ucrInputFootnotesMultiline.SetItems(dctFootnotesMultiline)
+        ucrChkFootnotesMultiline.AddParameterPresentCondition(True, "footnotes.marks")
+        ucrChkFootnotesMultiline.AddParameterPresentCondition(False, "footnotes.marks", False)
```
are these the correct settings for multiline?

lloyddewit reviewed (changes requested):
@anastasia-mbithe Thank you, this PR needed a lot of extra code and it all looks neatly and professionally implemented.
My comments are mainly small aesthetic things.

anastasia-mbithe commented:
@lloyddewit, I made the changes

lloyddewit commented:
@rdstern Please could you retest the multiline (the only code that functionaly changed), we can then merge, thanks

rdstern reviewed (dismissed):
@anastasia-mbithe and @lilyclements this all seems to work, but I assume it will need to change now to get rid of mmtable2?

But perhaps not?  When the main tables command is merged, then perhaps the mmtable2 disappears automatically.  I am approving, but could lily also approve before it is merged.

rdstern commented:
@lloyddewit I checked with @lilyclements and would be happy to see this issue merged.

anastasia-mbithe commented:
@rdstern and @lloyddewit, This branch had some conflicts from the merging of the other related PR. The conflicts are resolved now, but that has dismissed your reviews/approvals.
